dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathEntry.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+/** An entry in Java class path. */
+interface ClassPathEntry {
```

Comment:
Why interface instead of abstract class? Abstract classes are more powerful and much more changeable. E.g. you can add methods to one without breaking all subclasses.

Reply:
I made this as as interface because I don't need any concrete methods or fields. Interfaces can have implementation too. I don't see one is better than the other in this case.

Reply:
This point might become moot, but I really think the default should be class, not interface, without a strong reason otherwise. Interfaces are much more brittle than classes and have less functionality. They are vastly overused in Java today to no good end.

Reply:
Changed this from interface to a concrete class.